lib/ansible/modules/cloud/vmware/vmware_guest_customization.py

```diff
@@ -0,0 +1,563 @@
+        try:
+            current_spec = self.cc_mgr.GetCustomizationSpec(name=self.spec_name)
+        except vim.fault.NotFound as e:
+            self.module.fail_json(msg="Failed to retrieve customization"
+                                      " specification '%s' while updating : %s" % (self.spec_name,
+                                                                                   to_native(e.msg)))
+        spec_type = self.params.get('spec_type')
+        if spec_type != current_spec.info.type:
+            self.module.fail_json(msg="Can not change the type of customization"
+                                      " specification from '%s' to '%s'" % (spec_type,
+                                                                            current_spec.info.type))
+
+        new_spec_name = self.params.get('new_spec_name')
+        if new_spec_name and new_spec_name != current_spec.info.name:
+            if self.cc_mgr.DoesCustomizationSpecExist(name=new_spec_name):
+                self.module.fail_json(msg="Unable to rename customization specification. Please specify"
+                                          " different name for customization spec as '%s' already exists"
+                                          " in given configuration." % new_spec_name)
+            if not self.module.check_mode:
+                self.cc_mgr.RenameCustomizationSpec(name=self.spec_name, newName=new_spec_name)
+            self.module.exit_json(changed=True)
+
+        reconf_flag = False
+        new_description = self.params.get('spec_description')
+        if new_description and current_spec.info.description != new_description:
+            current_spec.info.description = new_description
+            reconf_flag = True
+
+        new_hostname = self.params.get('vm_hostname')
+        if new_hostname and current_spec.spec.identity.hostName.name != new_hostname:
+            current_spec.spec.identity.hostName.name = new_hostname
+            reconf_flag = True
+
+        new_domain = self.params.get('domain', None)
+        if new_domain and current_spec.spec.identity.doamin != new_domain:
+            current_spec.spec.identity.domain = new_domain
+            reconf_flag = True
+
+        new_timezone = self.params.get('timezone')
+        if new_timezone and current_spec.spec.identity.timeZone != new_timezone:
+            current_spec.spec.identity.timeZone = new_timezone
+            reconf_flag = True
+
+        new_dns_servers = self.params.get('dns_servers')
+        if new_dns_servers and set(current_spec.globalIPSettings.dnsServerList) != set(new_dns_servers):
+            current_spec.globalIPSettings.dnsServerList = new_dns_servers
+            reconf_flag = True
+
+        new_dns_suffix = self.params.get('dns_suffix')
+        if new_dns_servers and set(current_spec.globalIPSettings.dnsSuffixList) != set(new_dns_suffix):
+            current_spec.globalIPSettings.dnsSuffixList = new_dns_suffix
+            reconf_flag = True
+
```

Comment:
I'm missing a bunch of items like network info, `joindomain` etc?

Reply:
I don't have Windows licensed machine to test this.